### PR TITLE
Add initial alsa support

### PR DESCRIPTION
**Pull request by tiopex**
Initial alsa support

### CODE REVIEW

**Comment by Apaczer** [comment]
It does output snd to alsa now! Did you test any game that have satisfactory sound quality?

I've started to refactor for -D__MIYOO_ specific changes. Could you rebase, mv to miyoo_sdk.c & add platform macros in main?

**Comment by tiopex** [comment]
If you run the binary from the BitRiders release, the ALSA settings (samples, periods, etc.) are printed to the console. After the correct setup, the sound should have fine quality.

However, for different ROMs, these ALSA settings vary. It must also have some kind of algorithm to determine these values.

**Comment by tiopex** [comment]
The sound quality should be fine now

**Comment by Apaczer** [comment]
Great! Tested only with SMetroid and with latest change the remaining issue have been adressed (echo sound).

**Comment by Apaczer** [comment]
About "compatible"mode  (which is just a binary compiled with SA1 "Super FX assembler" from snes9x), ~~I should mentiod that @bitrider78 was babbling to the end about superior performance on ASM *_fast binary (Assembly library of spc700 cpu)~~ but either way he decided to stick with "compatible" on final release. We might just end up using SA1 in the end.

EDIT:
Right he ment APU compatible so SA1 for that reason,

File: src/menu.c [hunk 1]
```diff
@@ -2264,10 +2264,9 @@ int MainMenu(int prevaction)
 	gp_initGraphics(16,currFB,1);
 	gp_clearFramebuffer16((unsigned short*)framebuffer16[currFB],0x0);
 	MenuFlip();
-#if !defined(__WIZ__)
+#if !defined(__WIZ__) && !defined(__MIYOO__)
```
Comment (Apaczer) [inline src/menu.c:2267]: yikes, slept over that one. Thanks!

**Pull request merged**